### PR TITLE
Minor refactor in ApplicationController

### PR DESCRIPTION
**Pull request by seabl**
No issue opened

* [x] PR is descriptively titled 📑 
* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] ask `@publiclab/reviewers` for help, in a comment below

Thanks!

### CODE REVIEW

**Comment by seabl** [comment]
@publiclab/reviewers Could someone please give a review. Thanks.

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -72,9 +72,7 @@ def prompt_login(message = I18n.t('application_controller.must_be_logged_in'))
   end
 
   def current_user_session
-    return @current_user_session if defined?(@current_user_session)
-
-    @current_user_session = UserSession.find
+    @current_user_session ||= UserSession.find
```
Comment (Uzay-G) [inline app/controllers/application_controller.rb]: This part is good. Thanks!

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -202,15 +200,4 @@ def signed_in?
   def page_not_found
     render file: "#{Rails.root}/public/404.html", layout: false, status: :not_found
   end
-
```
Comment (Uzay-G) [inline app/controllers/application_controller.rb]: Why did you delete this part?

Comment (seabl) [reply app/controllers/application_controller.rb]: Because there's another function of the same name in application helpers, which I myself refactored in a previous PR. I've now moved that function here, to fix the test errors.

Comment (Uzay-G) [reply app/controllers/application_controller.rb]: Alright this seems good! I will approve once travis passes

Comment (seabl) [reply app/controllers/application_controller.rb]: It got worse :( I'm reverting back to 1st commit on this brach.

Comment (Uzay-G) [reply app/controllers/application_controller.rb]: It failed because you removed the `logged_in_as` method for the application helper. 
Maybe, on this pull request, just edit the current_user_session method and don't change the existing 'log_in_as' methods

Comment (Uzay-G) [reply app/controllers/application_controller.rb]: Oh yeah alright you did it :+1:

Comment (seabl) [reply app/controllers/application_controller.rb]: Yes, that's the reason. But now, we still have 2 `logged_in_as` functions. Maybe that will have to get its own PR. At least, now the PR title is more accurate ;)

**Comment by seabl** [comment]
@Uzay-G Thank you for the review.

**Comment by Uzay-G** [comment]
No problem :+1:

**Comment by SidharthBansal** [comment]
🎉

**Comment by seabl** [comment]
:+1:
